### PR TITLE
Docs secure endpoints

### PR DESCRIPTION
**Pull request by robnewman**
# Description
Outline of how to establish a secure data perimeter (with links to all relevant AWS services and guides)

### CODE REVIEW

File: docs/advanced-features/private-endpoint-access.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+Establishing a **data perimeter** pattern that only allows access to trusted
+principals from trusted networks is a best practice to guarantee your
+organization's data security. A data perimeter helps protect your data
+from unintended access and potential configuration errors via
+built-in barriers.
+
+You can restrict Amazon S3 bucket access to a particular VPC and VPN traffic
+via a data perimeter pattern, which prevents leaked S3 credentials from 
+bypassing your organizations VPN.
```
Comment (akarve) [inline docs/advanced-features/private-endpoint-access.md:14]:
```suggestion
bypassing your organization's VPN.
```

**Comment by robnewman** [comment]
@sir-sigurd Aneesh and I both want you to review these docs carefully before we ship them. Please feel free to add whatever is needed to make it clear what is needed to be done by the customer to get a secure data perimeter deployed. Thank you!

File: docs/advanced-features/private-endpoint-access.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+The example policy below denies access (`"Effect": "Deny"`) to all
+Amazon S3 buckets prefixed with the string `quilt`
+unless **all** of the following conditions are met:
+
+1. A `Source VPN` matches either `vpc-LOCAL` or `vpc-VPN`.
+2. A tag attached to the principal making the request with the key
+`aws:PrincipalTag/NetworkRestrictedExempt` has the value `true`
+3. The request must come from any IP **except** the range 
+`192.0.2.0 - 192.0.2.255` and `203.0.113.0 - 203.0.113.255`
```
Comment (sir-sigurd) [inline docs/advanced-features/private-endpoint-access.md:53]:
```suggestion
3. The request comes from IP range `192.0.2.0 - 192.0.2.255` or `203.0.113.0 - 203.0.113.255`.
```

**Review by sir-sigurd — COMMENTED** [review]
I'm not 100% sure with my suggestions to explanation of what example policy does.

**Comment by robnewman** [comment]
All changes requested by @sir-sigurd have been committed. Sergey is going to add instructions for what customers need to do (ask for a new variant) and how to confirm everything is correctly set up, then we should be complete.

File: docs/advanced-features/private-endpoint-access.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+You can restrict Amazon S3 bucket access to a particular VPC and VPN traffic
+via a data perimeter pattern, which prevents leaked S3 credentials from 
+bypassing your organization's VPN.
+
+> Quilt already has private IPs for all Quilt services (Lambda
+functions, API Gateway, Quilt catalog API).
+
+To implement a data perimeter, you will need to take the following steps.
+
+## 1. Configure and deploy a Service Control Policy and Amazon S3 bucket policy
```
Comment (sir-sigurd) [inline docs/advanced-features/private-endpoint-access.md:21]: Probably this item should go last because it requires S3 gateway endpoint.

Comment (robnewman) [reply docs/advanced-features/private-endpoint-access.md]: Good catch. Done.

**Review by sir-sigurd — APPROVED** [review]

**Pull request merged**